### PR TITLE
MNT: Accumulated Convention Updates

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

I went around and updated every out of date CI/structural file that I noticed, including moving the tests into an inner submodule.

## Motivation and Context

It is finally time for nabs to rise again.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        exclude: '^(conda-recipe/meta.yaml)$'
+    -   id: debug-statements
+
+-   repo: https://gitlab.com/pycqa/flake8.git
+    rev: 3.8.3
+    hooks:
+    -   id: flake8
+
+-   repo: https://github.com/timothycrosley/isort
+    rev: 5.5.2
```
Comment (klauer) [inline .pre-commit-config.yaml:26]:
```suggestion
    rev: 5.6.4
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    -   id: check-json
+    -   id: check-merge-conflict
+    -   id: check-symlinks
+    -   id: check-xml
+    -   id: check-yaml
+        exclude: '^(conda-recipe/meta.yaml)$'
+    -   id: debug-statements
+
+-   repo: https://gitlab.com/pycqa/flake8.git
+    rev: 3.8.3
```
Comment (klauer) [inline .pre-commit-config.yaml:21]:
```suggestion
    rev: 3.8.4
```

**Comment by ZLLentz** [comment]
I'll return to this later today (hopefully) to figure out which existing methods to keep and get passing in the tests

File: conda-recipe/meta.yaml [hunk 3]
```diff
@@ -1,27 +1,30 @@
...
     build:
-      - python
+      - python >=3.6
+      - setuptools
 
     run:
-      - bluesky 
-      - python
+      - python >=3.6
+      - bluesky
```
Comment (klauer) [inline conda-recipe/meta.yaml:21]: Looks like pandas, lmfit, scipy may be missing

Handy alias: `git grep -h 'import ' |sort |uniq`

File: docs/release_notes.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import sys
```
Comment (klauer) [inline docs/release_notes.py:1]: Should upstream this to `pcdsutils` one day (but not today)

File: nabs/tests/test_streams.py [hunk 5]
```diff
@@ -1,9 +1,23 @@
 from bluesky.callbacks import CallbackCounter
-from bluesky.examples import stepscan
 from bluesky.tests.utils import DocCollector
+from bluesky.utils import Msg
+
 from nabs.streams import AverageStream
 
 
+def stepscan(det, motor):
+    """Old example scan from bluesky deprecated submodule."""
```
Comment (klauer) [inline nabs/tests/test_streams.py:9]: Nice 😁

File: nabs/tests/test_plan_stubs.py [hunk 6]
```diff
@@ -16,6 +16,7 @@ def test_measure_average(RE, hw):
     def measure_plan(detectors):
         yield from open_run()
         ret = yield from measure_average(detectors, num=250)
+        logger.debug(ret)
```
Comment (klauer) [inline nabs/tests/test_plan_stubs.py:19]: `ret` isn't a string, is it?

**Review by klauer — APPROVED** [review]
Looks good overall - NABS here we come!

Oh, one logging bug may be getting hidden by pytest's logging/stderr capture.

File: conda-recipe/meta.yaml (hunk 3 above)
Comment (ZLLentz) [reply conda-recipe/meta.yaml:21]: I don't see pandas or lmfit anywhere in the repo, but I do see the missing scipy dependency

File: docs/release_notes.py (hunk 4 above)
Comment (ZLLentz) [reply docs/release_notes.py:1]: Yes

File: nabs/tests/test_plan_stubs.py (hunk 6 above)
Comment (ZLLentz) [reply nabs/tests/test_plan_stubs.py:19]: `ret` is a `dict` here, does that cause issues?

File: conda-recipe/meta.yaml (hunk 3 above)
Comment (klauer) [reply conda-recipe/meta.yaml:21]: I was on the wrong branch locally - my apologies (`pswalker_upstream` from #22)

Comment (ZLLentz) [reply conda-recipe/meta.yaml:21]: Ah! That makes sense

File: nabs/tests/test_plan_stubs.py (hunk 6 above)
Comment (klauer) [reply nabs/tests/test_plan_stubs.py:19]: I thought it wasn't allowed, but apparently am wrong here as well:

```
In [1]: import logging

In [3]: logging.error({})
ERROR:root:{}

In [4]: logging.error([])
ERROR:root:[]

In [5]: logging.error(['a', 'b', 'c'])
ERROR:root:['a', 'b', 'c']
```

It is a bit weird, though, I think...

Comment (ZLLentz) [reply nabs/tests/test_plan_stubs.py:19]: I'll revise to build a clear log message here then, this one was helpful for debugging

**Pull request merged**